Gems/Multiplayer/Code/Source/Pipeline/NetworkPrefabProcessor.cpp

```diff
@@ -53,143 +80,115 @@ namespace Multiplayer
+                ->Version(6)
             ;
         }
     }
 
-    static void GatherNetEntities(
-        AzToolsFramework::Prefab::Instance* instance,
-        AZStd::unordered_map<AZ::Entity*, AzToolsFramework::Prefab::Instance*>& entityToInstanceMap,
-        AZStd::vector<AZ::Entity*>& netEntities)
+    static void PopulateNetworkInstance(
+        AzToolsFramework::Prefab::Instance* sourceInstance,
+        AzToolsFramework::Prefab::Instance* networkInstance,
+        AzToolsFramework::Prefab::Instance* rootSourceInstance,
+        AzToolsFramework::Prefab::Instance* rootNetworkInstance,
+        const AZStd::string& prefabName,
+        const AZStd::string& uniqueName,
+        AzToolsFramework::Prefab::PrefabConversionUtils::PrefabProcessorContext& context,
+        AzFramework::Spawnable* networkSpawnable)
     {
-        instance->GetEntities([instance, &entityToInstanceMap, &netEntities](AZStd::unique_ptr<AZ::Entity>& prefabEntity)
+        using AzToolsFramework::Prefab::PrefabConversionUtils::ProcessedObjectStore;
+        using namespace AzToolsFramework::Prefab;
+
+        sourceInstance->GetEntities([networkInstance, rootSourceInstance, rootNetworkInstance, prefabName, uniqueName, &context, networkSpawnable](AZStd::unique_ptr<AZ::Entity>& sourceEntity)
         { 
-            if (prefabEntity->FindComponent<NetBindComponent>())
+            if (sourceEntity->FindComponent<NetBindComponent>())
             {
-                AZ::Entity* entity = prefabEntity.get();
-                entityToInstanceMap[entity] = instance;
-                netEntities.push_back(entity);
+                AZ::Entity* entity = sourceEntity.get();
+                AZ::Entity* netEntity = SpawnableUtils::CreateEntityAlias(
+                    prefabName,
+                    *rootSourceInstance,
+                    uniqueName,
+                    *rootNetworkInstance,
+                    *networkInstance,
+                    entity->GetId(),
+                    AzToolsFramework::Prefab::PrefabConversionUtils::EntityAliasType::Replace,
+                    AzToolsFramework::Prefab::PrefabConversionUtils::EntityAliasSpawnableLoadBehavior::DependentLoad,
+                    NetworkEntityManager::NetworkEntityTag,
+                    context);
+
+                AZ_Assert(
+                    netEntity, "Unable to create alias for entity %s [%zu] from the source prefab instance", entity->GetName().c_str(),
+                    aznumeric_cast<AZ::u64>(entity->GetId()));
+
+                netEntity->InvalidateDependencies();
+                netEntity->EvaluateDependencies();
+
+                Multiplayer::PrefabEntityId prefabEntityId;
+                prefabEntityId.m_prefabName = sourceEntity->GetName();
+                netEntity->FindComponent<NetBindComponent>()->SetPrefabAssetId(networkSpawnable->GetId());
+                netEntity->FindComponent<NetBindComponent>()->SetPrefabEntityId(prefabEntityId);
```

Comment:
You can just grab `NetBindComponent` once here and then call the sets.

Reply:
Fixed!